week4-ts/my-app/src/components/SearchResult/UserCard.tsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import styled from "styled-components";
+import { Link } from "react-router-dom";
+import { User } from "../../types";
+
+interface Props {
+  user: User;
+}
+
+function UserCard({ user }: Props) {
+  return (
+    <UserContainer>
+      <CloseBtn>
+        <Link to="/search"> X </Link>
+      </CloseBtn>
+      <UserImage src={user.avatar_url} alt="프로필사진" />
+      <UserName>{user.name}</UserName>
+      <UserId>{user.login}</UserId>
+      <UserLink
+        type="button"
+        onClick={() => {
+          window.open(user.html_url);
+        }}
+      >
+        깃허브 구경가기
+      </UserLink>
+      <UserInfoContainer>
+        <UserInfo>
+          <h1>{user.public_repos}</h1>
+          <p>Repos</p>
+        </UserInfo>
+        <UserInfo>
+          <h1>{user.followers}</h1>
+          <p>Followers</p>
+        </UserInfo>
+        <UserInfo>
+          <h1>{user.following}</h1>
+          <p>Following</p>
+        </UserInfo>
+      </UserInfoContainer>
+    </UserContainer>
+  );
+}
```

Comment:
```suggestion ts
function UserCard({ user }: Props) {
const { avatar_url, name, login, html_url, public_repos, followers, following } = user;

  return (
      <UserContainer>
      <CloseBtn>
        <Link to="/search"> X </Link>
      </CloseBtn>
      <UserImage src={avatar_url} alt="프로필사진" />
      <UserName>{name}</UserName>
      <UserId>{login}</UserId>
      <UserLink
        type="button"
        onClick={() => {
          window.open(html_url);
        }}
      >
        깃허브 구경가기
      </UserLink>
      <UserInfoContainer>
        <UserInfo>
          <h1>{public_repos}</h1>
          <p>Repos</p>
        </UserInfo>
        <UserInfo>
          <h1>{followers}</h1>
          <p>Followers</p>
        </UserInfo>
        <UserInfo>
          <h1>{following}</h1>
          <p>Following</p>
        </UserInfo>
      </UserInfoContainer>
    </UserContainer>
  );
}
```
로 구조분해 할당하면 좀더깔끔해집니다!

Reply:
버튼 타입도 지정하자!!!

Reply:
오왁!! 구조분해할당을 앞으로 적극적으로 사용해볼께여!!!